frame/alliance/src/lib.rs

```diff
@@ -604,25 +615,46 @@ pub mod pallet {
 		}
 
 		/// Initialize the founders, fellows, and allies.
+		/// Resets members and removes all active proposals if alliance is already initialized.
 		///
 		/// This should only be called once, and must be called by the Root origin.
-		#[pallet::weight(T::WeightInfo::init_members(
+		#[pallet::weight(T::WeightInfo::force_set_members(
 			T::MaxFounders::get(),
 			T::MaxFellows::get(),
 			T::MaxAllies::get()
 		))]
-		pub fn init_members(
+		pub fn force_set_members(
 			origin: OriginFor<T>,
 			founders: Vec<T::AccountId>,
 			fellows: Vec<T::AccountId>,
 			allies: Vec<T::AccountId>,
 		) -> DispatchResult {
 			ensure_root(origin)?;
 
-			// Cannot be called if the Alliance already has Founders or Fellows.
-			// TODO: Remove check and allow Root to set members at any time.
-			// https://github.com/paritytech/substrate/issues/11928
-			ensure!(!Self::is_initialized(), Error::<T, I>::AllianceAlreadyInitialized);
+			if Self::is_initialized() {
+				// Reset Alliance by removing all members.
+				// Veto and remove all active proposals to avoid any unexpected behavior from
+				// actionable items managed outside of the pallet. Items like `UnscrupulousWebsites`
+				// managed within the pallet left for new Alliance to cleanup or keep.
+
+				T::ProposalProvider::proposals().into_iter().for_each(|hash| {
+					T::ProposalProvider::veto_proposal(hash);
+				});
+
+				T::MembershipChanged::change_members_sorted(&[], &Self::votable_members(), &[]);
+				ensure!(
+					// `MemberRole` variants, the key of the `StorageMap`, is not expected to grow
+					// to 255.
+					Members::<T, I>::clear(255, None).maybe_cursor.is_none(),
+					Error::<T, I>::AllianceNotReset
```

Comment:
I'm not the most familiar with the new `clear` functions, but _if_ there were more than 255 members, would it still clear the first 255? In that case, it would still have done something useful, and maybe this could first emit an event, like `Event::MembersRemoved{ count: 255 }` and then error with `Error::UnableToRemoveAllMembers`.

Reply:
Or does the prefix refer to variants, i.e. `Founder, Fellow, Ally, Retiring`? In that case I think we can safely say that it won't be above 255. I just don't want to make the assumption that members would be limited to 255.

Reply:
Yes, the key of the `Members` storage prefix is variants from `MemberRole` enum. Here we assume there is no more than 255 roles.